src/sagemaker_xgboost_container/algorithm_mode/callback.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+import tempfile
+import threading
+import queue
+from typing import Callable, Optional
+from xgboost import rabit
+from xgboost.core import CallbackEnv
+from xgboost.callback import _fmt_metric as fmt_metric
+
+
+TEMP_FILE_SUFFIX = ".sagemaker-ignore"
+FILE_LOCK_SUFFIX = ".sagemaker-uploading"
+FILE_SAFE_SUFFIX = ".sagemaker-uploaded"
+
+CHECKPOINT_FILENAME = "xgboost-checkpoint"
+CHECKPOINT_NUM_DIGITS = 12
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def _zero_pad(iteration: int) -> str:
+
+    iter_num = format(iteration, "0{}".format(CHECKPOINT_NUM_DIGITS))
+
+    return iter_num
+
+
+# modified from https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/callback.py
+def print_evaluation(period=1, show_stdv=True, start_iteration=0):
+    """Create a callback that print evaluation result.
+
+    We print the evaluation results every **period** iterations
+    and on the first and the last iterations.
+
+    Parameters
+    ----------
+    period : int
+        The period to log the evaluation results
+
+    show_stdv : bool, optional
+         Whether show stdv if provided
+
+    Returns
+    -------
+    callback : function
+        A callback that print evaluation every period iterations.
+    """
+    def callback(env):
+        """internal function"""
```

Comment:
Will add in the next PR, which will be a separate PR because I don't have cross-repository auto-update of this PR as I commented in the general comment. Adding last-minute comments before closing this PR.